src/components/Identity.vue

```diff
@@ -233,6 +250,23 @@ export default {
     },
   },
   methods: {
+    rearrangeSelf(e) {
```

Comment:
I really like this approach to implementing keyboard-based D&D. Submitting the necessary events back through the same event system should give a really consistent bit of user interface, without a huge amount of code duplication. How does this feel in practice?

Reply:
Thank you! I was doing my best to fight code duplication the whole way with this, and I think it worked out; I think it does feel pretty ok in practice, actually, with my one worry being the usual one about making sure both the emitter and the listener (in IdentityForm.vue) know what the schema for the custom event object should look like; I need to write some tests to try to make sure that they're both emitting/expecting an object with the same properties.

Reply:
Yeah, tests for this whole system would be amazing. I know what you mean about anxiety that the emitted and received events have the same object schema; this is something that's more likely than most things are to break during, say, a Vue upgrade down the line.

Spending that anxiety on tests is a really healthy and productive reaction. I don't have a good handle on how to test Vue components, but a brief search on Google turns up a couple of promising approaches.

Reply:
There are already many tests for the Vue system present in `/tests/unit`; I wrote a few tests over in #128, based on those, which I think are working out; I was going to have to add tests for this too, because according to the readme both server and client side code is required to have 100% test coverage in this repository (although it may in fact be impossible to get the code that affects scroll behavior when the components are moved off-screen to be properly tested in a virtual DOM set up like the existing one, so I may have to ask for an exception)